test/necb/unit_tests/tests/test_necb_structures.rb

```diff
@@ -63,13 +63,86 @@ def test_necb_structures()
       @buildings.sort.each   do |building |
         @templates.sort.each do |template |
           cas = "CASE #{building} (#{template})"
+          tag = "space_conditioning_category"
 
           st    = Standard.build(template)
           model = st.model_create_prototype_model(template: template,
                                                   epw_file: epw,
                                                   building_type: building,
+                                                  construction_opt: 'structure',
                                                   sizing_run_dir: @sizing_run_dir)
 
+          attics  = []
+          plenums = []
+
+          model.getSpaces.each do |space|
+            prop = space.additionalProperties.getFeatureAsString(tag)
+            next if prop.empty?
+            next if space.partofTotalFloorArea
+
+            prop = prop.get.downcase
+            attics  << space if prop == "unconditioned"
+            plenums << space if prop == "nonresconditioned"
+          end
+
+          attics.each do |attic|
+            id  = attic.nameString
+            set = attic.defaultConstructionSet
+            err_msg = "BTAP::Structure #{id} default construction set (#{cas})?"
+            refute_empty(set, err_msg)
+
+            set = set.get
+            id  = set.nameString
+            err_msg = "BTAP::Structure default construction set #{id} (#{cas})?"
+            assert(id.include?("ATTIC"), err_msg)
+
+            attic.surfaces.each do |surface|
+              id = surface.nameString
+              c  = surface.construction.get.to_LayeredConstruction.get
+              next unless c.layers.size == 2
+              next unless surface.surfaceType.downcase == "floor"
+
+              if id.include?("soffit")
+                err_msg = "BTAP::Structure #{id} insulated (#{cas})?"
+
+                # Soffit 'floor' not insulated.
+                c.layers.each do |layer|
+                  assert(layer.nameString.include?("material"), err_msg)
+                end
+              else
+                id = c.layers.last.nameString
+                err_msg = "BTAP::Structure #{id} insulation layer (#{cas})?"
+                assert(id.include?("cellulose"), err_msg)
```

Comment:
Attic _interzone_ floors are insulated, while attic _exposed_ floors (i.e. soffits) do not - good.